### ISSUE TITLE
Using off-center focal point can result in incorrectly sized images

### ISSUE THREAD

**Issue by waqasakramdot** [dotCMS/core#24052]
### Problem Statement

 when using the focal point with cropping parameters, if the focal point is close enough to the edge of the image, the image is cut off

https://demo.dotcms.com/contentAsset/image/d331052e-8220-42d8-a4e4-3ea80e088123/fileAsset/filter/Crop/crop_w/600/crop_h/600/fp/0.7,0.6

### Steps to Reproduce

The original image's size is 1920x700, and the URL specifies a cropping area of 600x600 with a focal point of x: 0.7, y = 0.6. The output size is 600x579, instead of 600x600. If you adjust the Y focal point to be closer to center, the image size starts getting closer and closer to the correct size, but not until the Y focal point is altered to 0.56.

What I found is that the resulting cropping area's coordinates can extend beyond the image's bounds sometimes, and the result is an image of the incorrect size

![image](https://user-images.githubusercontent.com/102758033/217324235-7d30a9db-5431-4dca-acc5-1e37151b8922.png)

### Acceptance Criteria

We discussed with the customer about possible 4 potential outcomes.

1. Crop as much as you can (as we are doing currently)
2. Add whitespace to make the image the correct size
3. Move the focal point to a spot that allows for proper cropping
4. Throw an error

Ans: Result 3 would be the most ideal. we would rather have a correctly sized image that isn't perfectly centered on a focal point than an incorrectly sized image, added whitespace, or no image at all. A nice to have would be a warning in the log when this situation occurs 

### dotCMS Version

23.1

### Proposed Objective

Customer Support

### Proposed Priority

Priority 4 - Trivial

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/109917
